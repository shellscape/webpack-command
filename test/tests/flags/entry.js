const { apply, build, crcDist, prep, test, validate } = require('../../util');
const parseEntries = require('../../../lib/entry');

test('--entry', module, () => {
  it('should parse input', () => {
    const entries = [fixturePath('common/entry-a.js'), fixturePath('common/entry-c.js')];
    const result = parseEntries({ entries, flags: {} });

    expect(result.main).toHaveLength(2);
    expect(result).toMatchSnapshot();
  });

  it('should parse single flag', () => {
    const flags = { entry: fixturePath('common/entry-a.js') };
    const result = parseEntries({ entries: [], flags });

    expect(result.main).toEqual(flags.entry);
    expect(result).toMatchSnapshot();
  });

  it('should parse flag array', () => {
    const flags = {
      entry: [fixturePath('common/entry-a.js'), fixturePath('common/entry-b.js')]
    };
    const result = parseEntries({ entries: [], flags });

    expect(result.main).toHaveLength(2);
    expect(result).toMatchSnapshot();
  });

  it('should parse flag object', () => {
    const flags = {
      entry: {
        a: fixturePath('common/entry-a.js'),
        b: fixturePath('common/entry-b.js')
      }
    };
    const result = parseEntries({ entries: [], flags });

    expect(Object.keys(result)).toHaveLength(2);
    expect(result).toMatchSnapshot();
  });

  it('should parse input + single flag', () => {
    const entries = [fixturePath('common/entry-a.js')];
    const flags = { entry: fixturePath('common/entry-b.js') };
    const result = parseEntries({ entries, flags });

    expect(result.main).toHaveLength(2);
    expect(result).toMatchSnapshot();
  });

  it('should parse input + flag array', () => {
    const entries = [fixturePath('common/entry-a.js')];
    const flags = {
      entry: [fixturePath('common/entry-b.js'), fixturePath('common/entry-c.js')]
    };
    const result = parseEntries({ entries, flags });

    expect(result.main).toHaveLength(3);
    expect(result).toMatchSnapshot();
  });

  it('should parse input + flag object', () => {
    const entries = [fixturePath('common/entry-a.js')];
    const flags = {
      entry: {
        a: fixturePath('common/entry-a.js'),
        b: fixturePath('common/entry-b.js')
      }
    };
    const result = parseEntries({ entries, flags });

    expect(typeof result.main).toBe('string');
    expect(Object.keys(result)).toHaveLength(3);
    expect(result).toMatchSnapshot();
  });

  for (const name of ['multi' /* , 'single'*/]) {
    const fixture = `entry/${name}`;
    const opts = { fixture };
    let config;

    it(`${name} should validate`, () => {
      expect(validate(opts)).toEqual(true);
    });

    it(`${name} should apply`, () => {
      const { argv } = prep(opts);
      const entry = parseEntries({ entries: [], flags: argv });

      config = apply(opts);

      if (entry) {
        config.entry = entry;
      }

      expect(config).toMatchSnapshot();
    });

    it(`${name} should build`, () =>
      build(config).then((result) => {
        expect(result).toMatchSnapshot();
        expect(crcDist()).toMatchSnapshot();
      }));
  }
});
