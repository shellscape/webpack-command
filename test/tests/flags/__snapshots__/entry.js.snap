// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --entry > multi should apply #0 1`] = `
Object {
  "entry": Object {
    "main": Array [
      "./test/fixtures/common/entry-a.js",
      "./test/fixtures/common/entry-b.js",
    ],
  },
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --entry > multi should build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  4.94 KiB    main  [emitted]  main
Entrypoint main = main.js
[0] multi ./test/fixtures/common/entry-a.js ./test/fixtures/common/entry-b.js 40 bytes {main} [built]
[./test/fixtures/common/entry-a.js] 40 bytes {main} [built]
[./test/fixtures/common/entry-b.js] 40 bytes {main} [built]"
`;

exports[`Flags > --entry > multi should build #1 1`] = `"73fbdb7c"`;

exports[`Flags > --entry > should parse flag array #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
  ],
}
`;

exports[`Flags > --entry > should parse flag object #0 1`] = `
Object {
  "a": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "b": "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
}
`;

exports[`Flags > --entry > should parse input #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-c.js",
  ],
}
`;

exports[`Flags > --entry > should parse input + flag array #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-c.js",
  ],
}
`;

exports[`Flags > --entry > should parse input + flag object #0 1`] = `
Object {
  "a": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "b": "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
  "main": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
}
`;

exports[`Flags > --entry > should parse input + single flag #0 1`] = `
Object {
  "main": Array [
    "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
    "<PROJECT_ROOT>/test/fixtures/common/entry-b.js",
  ],
}
`;

exports[`Flags > --entry > should parse single flag #0 1`] = `
Object {
  "main": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
}
`;
